cluster-autoscaler/metrics/liveness_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func getTestResponse(timeout time.Duration, checkMonitoring bool) *httptest.ResponseRecorder {
+	req := httptest.NewRequest("GET", "/health-check", nil)
+	w := httptest.NewRecorder()
+	healthCheck := NewHealthCheck(timeout)
+	if checkMonitoring {
```

Comment:
you should explicitly set the lastActivity field. Then you won't need to rely on nanoseconds.